protobuf/lib/src/protobuf/coded_buffer_writer.dart

```diff
@@ -420,23 +420,14 @@ class CodedBufferWriter {
 
   /// Copy bytes from the given typed data array into the output buffer.
   ///
-  /// Has a specialization for Uint8List for performance.
-  int _copyInto(Uint8List buffer, int pos, TypedData value) {
-    if (value is Uint8List) {
-      var len = value.length;
-      for (var j = 0; j < len; j++) {
-        buffer[pos++] = value[j];
-      }
-      return pos;
-    } else {
-      var len = value.lengthInBytes;
-      var u8 = Uint8List.view(
-          value.buffer, value.offsetInBytes, value.lengthInBytes);
-      for (var j = 0; j < len; j++) {
-        buffer[pos++] = u8[j];
-      }
-      return pos;
-    }
+  /// Has a specialization for [Uint8List] for performance.
+  static int _copyInto(Uint8List buffer, int pos, TypedData value) {
```

Comment:
The code was originally written in this style to ensure good specialised loop for `Uint8List` to `Uint8List` copying. Using `setAll` does not yield the same good code unfortunately. 

I have confirmed that this is a regression both on `to_binary` and `query_encode_binary` benchmarks:

![plot](https://user-images.githubusercontent.com/131846/194511203-d0782b17-1cf3-4c94-8a48-8d99561ab4e4.png)
![plot-to-binary](https://user-images.githubusercontent.com/131846/194511205-fdb7949c-c131-4aab-9271-f9bb5896738a.png)

These are M1 measurements.

Reply:
Thanks! Closing.